pkg/engine/artifacts.go

```diff
@@ -199,14 +199,28 @@ func kubernetesAddonSettingsInit(profile *api.Properties) []kubernetesFeatureSet
 			kubernetesFeatureSetting{
 				"kubernetesmasteraddons-unmanaged-azure-storage-classes.yaml",
 				"azure-storage-classes.yaml",
-				profile.AgentPoolProfiles[0].StorageProfile != api.ManagedDisks,
+				profile.AgentPoolProfiles[0].StorageProfile != api.ManagedDisks && !profile.IsAzureStackCloud(),
 				profile.OrchestratorProfile.KubernetesConfig.GetAddonScript(DefaultAzureStorageClassesAddonName),
 			})
 		kubernetesFeatureSettings = append(kubernetesFeatureSettings,
 			kubernetesFeatureSetting{
 				"kubernetesmasteraddons-managed-azure-storage-classes.yaml",
 				"azure-storage-classes.yaml",
-				profile.AgentPoolProfiles[0].StorageProfile == api.ManagedDisks,
+				profile.AgentPoolProfiles[0].StorageProfile == api.ManagedDisks && !profile.IsAzureStackCloud(),
+				profile.OrchestratorProfile.KubernetesConfig.GetAddonScript(DefaultAzureStorageClassesAddonName),
+			})
+		kubernetesFeatureSettings = append(kubernetesFeatureSettings,
```

Comment:
instead of duplicating these lines, you could assign a var for the only string that changes: 
```
    unmanagedStorageClassYaml := "kubernetesmasteraddons-unmanaged-azure-storage-classes.yaml"
    managedStorageClassYaml :=  "kubernetesmasteraddons-managed-azure-storage-classes.yaml"

if profile.IsAzureStackCloud() {
    unmanagedStorageClassYaml = "kubernetesmasteraddons-unmanaged-azure-storage-classes-custom.yaml"
    managedStorageClassYaml =  "kubernetesmasteraddons-managed-azure-storage-classes-custom.yaml"
}

kubernetesFeatureSettings = append(kubernetesFeatureSettings,
			kubernetesFeatureSetting{
				unmanagedStorageClassYaml,
				"azure-storage-classes.yaml",
				profile.AgentPoolProfiles[0].StorageProfile != api.ManagedDisks,
				profile.AgentPoolProfiles[0].StorageProfile != api.ManagedDisks && !profile.IsAzureStackCloud(),
				profile.OrchestratorProfile.KubernetesConfig.GetAddonScript(DefaultAzureStorageClassesAddonName),
			})
		kubernetesFeatureSettings = append(kubernetesFeatureSettings,
			kubernetesFeatureSetting{
				managedStorageClassYaml,
				"azure-storage-classes.yaml",
				profile.AgentPoolProfiles[0].StorageProfile == api.ManagedDisks,
				profile.AgentPoolProfiles[0].StorageProfile == api.ManagedDisks && !profile.IsAzureStackCloud(),
				profile.OrchestratorProfile.KubernetesConfig.GetAddonScript(DefaultAzureStorageClassesAddonName),
			})
```

Reply:
we could even go one step further and define `StorageClassYaml` based on `profile.AgentPoolProfiles[0].StorageProfile == api.ManagedDisks`